core/src/pixelcolor/transparent_color.rs

```diff
@@ -155,4 +156,183 @@ pub trait AlphaColor: PixelColor {
+                let a1 = self.alpha() as u16;
+                let r1 = self.r() as u16 * a1 / Self::MAX_A as u16;
+                let g1 = self.g() as u16 * a1 / Self::MAX_A as u16;
+                let b1 = self.b() as u16 * a1 / Self::MAX_A as u16;
+                let a2 = Self::MAX_A as u16 - self.alpha() as u16;
+                let r2 = other.r() as u16 * a2 / Self::MAX_A as u16;
+                let g2 = other.g() as u16 * a2 / Self::MAX_A as u16;
+                let b2 = other.b() as u16 * a2 / Self::MAX_A as u16;
+                let r = (r1 + r2) as u8;
+                let g = (g1 + g2) as u8;
+                let b = (b1 + b2) as u8;
+                <$base_type>::new(r, g, b)
+            }
+        }
+
+        impl AlphaColor for $type {
+            fn alpha(&self) -> u8 {
+                (self.0 >> $a_pos) as u8 & Self::MAX_A
+            }
+
+            const MAX_A: u8 = ((1usize << $a_bits) - 1) as u8;
+        }
+
+        impl HasAlphaColor for $base_type {
+            type AlphaColor = $type;
+
+            fn with_alpha(self, alpha: u8) -> Self::AlphaColor {
+                $type(self.into_storage() | (alpha as $storage_type) << $a_pos)
+            }
+        }
+
+        impl RgbColor for $type {
+            fn r(&self) -> u8 {
+                #![allow(trivial_numeric_casts)]
+
+                (self.0 >> $r_pos) as u8 & Self::MAX_R
+            }
+
+            fn g(&self) -> u8 {
+                #![allow(trivial_numeric_casts)]
+
+                (self.0 >> $g_pos) as u8 & Self::MAX_G
+            }
+
+            fn b(&self) -> u8 {
+                #![allow(trivial_numeric_casts)]
+
+                (self.0 >> $b_pos) as u8 & Self::MAX_B
+            }
+
+            const MAX_R: u8 = ((1usize << $r_bits) - 1) as u8;
+            const MAX_G: u8 = ((1usize << $g_bits) - 1) as u8;
+            const MAX_B: u8 = ((1usize << $b_bits) - 1) as u8;
+
+            const BLACK: Self = Self::new(0, 0, 0, Self::MAX_A);
+            const RED: Self = Self::new(Self::MAX_R, 0, 0, Self::MAX_A);
+            const GREEN: Self = Self::new(0, Self::MAX_G, 0, Self::MAX_A);
+            const BLUE: Self = Self::new(0, 0, Self::MAX_B, Self::MAX_A);
+            const YELLOW: Self = Self::new(Self::MAX_R, Self::MAX_G, 0, Self::MAX_A);
+            const MAGENTA: Self = Self::new(Self::MAX_R, 0, Self::MAX_B, Self::MAX_A);
+            const CYAN: Self = Self::new(0, Self::MAX_G, Self::MAX_B, Self::MAX_A);
+            const WHITE: Self = Self::new(Self::MAX_R, Self::MAX_G, Self::MAX_B, Self::MAX_A);
+        }
+
+        impl From<$data_type> for $type {
+            fn from(data: $data_type) -> Self {
+                let data = data.into_inner();
+
+                Self(data & Self::ARGB_MASK)
+            }
+        }
+
+        impl From<$type> for $data_type {
+            fn from(color: $type) -> Self {
+                Self::new(color.0)
+            }
+        }
```

Comment:
The other color types only implement conversion from and to the `RawUx` types, which these types should also do.